tests/test_validity.py

```diff
@@ -0,0 +1,96 @@
+import json
+
+import jsonschema
+import pytest
+from jsonschema.exceptions import best_match
+from referencing import Registry, Resource
+
+STAC_VERSION = "1.0.0"
+PROJECT_ID = "https://stac-extensions.github.io/projection/v1.1.0/schema.json"
+ITEM_ASSETS_ID = "https://stac-extensions.github.io/item-assets/v1.0.0/schema.json"
+RASTER_BANDS_ID = "https://stac-extensions.github.io/raster/v1.1.0/schema.json"
+
+
+def test_core(stac_fixture, validator_fixture, stac_version="1.0.0"):
+    for stac_dict in stac_fixture:
+        assert validator_fixture.validate_core(stac_dict, stac_dict["type"], stac_version), f"{stac_dict.id} is invalid"
+
+
+def test_proj_extensions(
+    stac_fixture,
+    validator_fixture,
+    stac_version=STAC_VERSION,
+    extension_id=PROJECT_ID,
+):
+    for stac_dict in stac_fixture:
+        if "stac_extensions" in stac_dict and extension_id in stac_dict["stac_extensions"]:
+            validator_fixture.validate_extension(stac_dict, stac_dict["type"], stac_version, extension_id)
+
+
+def test_item_assets_extension(
```

Comment:
Is there a reason on why not to put the proj extension and item_asset extension tests together? as they are almost the same code.

maybe 
```
@pytest.mark.parametrize("extension_id", EXTENSIONS)
def test_extensions(stac_fixture, validator_fixture, extension_id, stac_version=STAC_VERSION):
    for stac_dict in stac_fixture:
        if "stac_extensions" in stac_dict and extension_id in stac_dict["stac_extensions"]:
            validator_fixture.validate_extension(stac_dict, stac_dict["type"], stac_version, extension_id)
```
with the EXTENSIONS being the list of extensions

Reply:
Good catch! Will implement this.